### PR TITLE
build(client-presence): type and lint fixes

### PR DESCRIPTION
**Pull request by jason-ha**
Minor production runtime changes with introduction of getOrCreateRecord helper and retyping of Object.entries/keys utilities.

### CODE REVIEW

**Comment by jason-ha** [comment]
> ## Code Coverage Summary
> ↓ packages.framework.presence.src:
> Line Coverage Change: -1.21%    Branch Coverage Change: -0.63%
> Baseline commit: [5abfa01](https://github.com/microsoft/FluidFramework/commit/5abfa015aff9d639d82830f3ad828324d5680bd7) Baseline build: [ 308954 ](https://dev.azure.com/fluidframework/10f9b53e-c7fd-4538-9fff-13cd088a436c/_build/results?buildId=308954) Happy Coding!!
> 
> ### Code coverage comparison check failed!!

Disabling the code coverage comparison check as it is counting comments and types for coverable lines and erroneously finding a coverage regression.

File: packages/framework/presence/src/internalUtils.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+type RequiredAndNotUndefined<T> = {
+	[K in keyof T]-?: Exclude<T[K], undefined>;
+};
+
+/**
+ * Object.entries retyped to preserve known keys and their types.
+ *
+ * @internal
+ */
+export const objectEntries = Object.entries as <T>(o: T) => KeyValuePairs<T>;
```
Comment (RishhiB) [inline packages/framework/presence/src/internalUtils.ts:28]: should these helpers be migrated to core-utils?

Comment (jason-ha) [reply packages/framework/presence/src/internalUtils.ts:28]: Maybe. I don't like to pre-emptively place utilities in a shared space without more than one use. Certainly, consider if these are generally useful as you look at fixing things. If they are, then we refactor.

**Review by WillieHabi — APPROVED** [review]
Looks good to me

**Pull request merged**